### DISCUSSION TITLE
How do you use nvim-cmp for the hover / docs?

### DISCUSSION THREAD

**Discussion by NullVoxPopuli** [hrsh7th/nvim-cmp#993] [Q&A]
maybe I don't know what this is called, but, atm, I have `<leader><Space>` bound to: 
```lua
        local function n(line) vim.cmd([[nnoremap ]] .. line) end

        n([[<leader><Space> :lua vim.lsp.buf.hover()<CR>]])
```

which looks like: 
![image](https://user-images.githubusercontent.com/199018/168509179-0219f774-1da1-4282-95ef-5449aadef382.png)

but there are a couple problems:
 - I don't know what the two overloads are
 - I don't know how to flip through the overloads' docs
 
 but with https://github.com/hrsh7th/cmp-nvim-lsp-signature-help
 I can get overload docs _within_ the parenthesis, which is good:
 
![image](https://user-images.githubusercontent.com/199018/168509302-4c565f04-c346-434c-aa76-b3a8233df9fe.png)

but I still have things to figure out:
 - how do I pull up these overloads in `normal` upon either `hover()` or `signature_help()`?
    - how do I pull up these docs while I'm typing the function? (or maybe, how to I keep it as is, but flip through overloads?)
     ![image](https://user-images.githubusercontent.com/199018/168509482-53db52aa-b5eb-4c37-97a3-93a1ba811a18.png)
    - how do I unify the styling of these docs (while I'm typing) and the docs that show when I activate `hover()`?
 - also how do change how the markdown is rendered? (in general the docs / markdown rendering in neovim seems super broken (or it's just not using markdown idk))
 
 Thanks!

**Comment by DasOhmoff** [comment]
I also have this issue. Were you able to fix it?

**Comment by DasOhmoff** [comment]
@hrsh7th Hey, do you maybe know the answer to this question?

**Comment by anuramat** [comment]
[noice](https://github.com/folke/noice.nvim/) works great

before:
![image](https://github.com/hrsh7th/nvim-cmp/assets/45295405/d78a74cd-09f9-4e3b-813b-c173207ae94a)

after:
![image](https://github.com/hrsh7th/nvim-cmp/assets/45295405/c335d26f-03db-413d-82ff-724e03d86e33)

relevant noice config part is `lsp.override.*`
